### ISSUE TITLE
Some pages lead to horizontal scroll on mobile

### ISSUE THREAD

**Issue by jakkan** [halogenica/beautifulhugo#79]
The math sample, big image sample and code sample on the demo site take up moro horizontal space than the viewport width and lead to horizontal scroll being needed. A better solution might be to scale the content to fit in the viewport width.

Tested on Samsung Galaxy S8 and One+ 5 in Chrome

**Comment by cite** [comment]
For my own site, I fixed this with a media query:

```
diff --git a/static/css/main.css b/static/css/main.css
index e726059..9eb8575 100644
--- a/static/css/main.css
+++ b/static/css/main.css
@@ -658,8 +658,10 @@ pre {
   font-size: 16px;
   line-height: 1.5em;
 }
-pre code {
-  overflow-x: scroll;
+@media only screen and (min-width: 768px) {
+  pre code {
+    white-space: pre;
+  }
 }
```

This will keep scroll bars on devices with enough screen width, and line break code on mobile devices with narrow screens. I don't have any clue about how to do this the right way, however :-(

**Comment by Focus** [comment]
@cite did you mean `white-space: pre-wrap;` ? I found that on your website which works, but the one you provided didn't

**Comment by cite** [comment]
@Focus Erm, nope. I just opened the page in Chrome, used the "Developer Tools" and to highlight one of the code blocks, and it's showing a `pre` for me, not `pre-wrap`.

**Comment by Focus** [comment]
@cite Weird, I see `pre-wrap`

![screen shot 2017-09-23 at 11 48 57](https://user-images.githubusercontent.com/93248/30771646-4bbc9a7e-a055-11e7-913a-63bb0da4da1e.png)

**Comment by cite** [comment]
@Focus Ah, so you minized the window before opening the dev tools, right? In that case, the width dependent CSS override wouldn't be executed anymore, that's why you see `pre-wrap`, not `pre` :-)

**Comment by dancwilliams** [comment]
@cite 

Thanks for this info.  Worked great for me!  I was having all sorts of scrolling issues with mobile devices.

Here is my code:

```
diff --git a/static/css/codeblock.css b/static/css/codeblock.css
index 4e52038..c5a904b 100644
--- a/static/css/codeblock.css
+++ b/static/css/codeblock.css
@@ -3,10 +3,11 @@
 pre {
   font-size: 16px;
   line-height: 1.5em;
 }
-pre code {
-  white-space: pre;
+@media only screen and (min-width: 768px) {
+  pre code {
+    white-space: pre;
+  }
 }
```

All is well now.  Fantastic!

**Comment by bankpeppy** [comment]
yeah,me to.my landscape always view half image on my samsung galaxy J7

**Comment by halogenica** [comment]
Is this still an issue? I'd like to fix this problem if possible. thanks :)

**Comment by halogenica** [comment]
I'm wondering if this got fixed with #204
